lib/Basics/process-utils.cpp

```diff
@@ -960,6 +961,31 @@ void TRI_CreateExternalProcess(char const* executable,
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Reads from the pipe of processes
+////////////////////////////////////////////////////////////////////////////////
+
+bool TRI_ReadPipe(ExternalProcess const* process,
+                  char* buffer,
+                  size_t &bufferSize) {
```

Comment:
```suggestion
                  size_t& bufferSize) {
```
I don't like this API. Why does it modify `bufferSize`? 
Typically you would call such filesystem APIs like this:
```
char buf[100];
memset(&buf[0], 0, sizeof(buf));

size_t lenRead = TRI_ReadPipe(..., &buf[0], sizeof(buf));
```
which now would not compile.

In other filesystem APIs, the return value would indicate how many bytes were read. Sometimes the return type is unsigned, and negative return values indicate errors. But modifying the buffer size does not look great because you can't call the function in a loop without making a copy of your buffer's size.